frontend/src/pages/Graph/SummaryPanelNodeTraces.tsx

```diff
@@ -174,18 +175,29 @@ class SummaryPanelNodeTracesComponent extends React.Component<Props, State> {
     };
 
     const d = this.props.nodeData;
-
-    const promise = d.workload
-      ? API.getWorkloadTraces(d.namespace, d.workload, params, d.cluster)
-      : d.service
-      ? API.getServiceTraces(d.namespace, d.service, params, d.cluster)
-      : API.getAppTraces(d.namespace, d.app!, params, d.cluster);
+    let perfKey: string;
+    let promise: Promise<any>;
+    switch (true) {
+      case !!d.workload:
+        perfKey = 'WorkloadTraces';
+        promise = API.getWorkloadTraces(d.namespace, d.workload!, params, d.cluster);
+        break;
+      case !!d.service:
+        perfKey = 'ServiceTraces';
+        promise = API.getServiceTraces(d.namespace, d.service!, params, d.cluster);
+        break;
+      default:
+        perfKey = 'AppTraces';
+        promise = API.getAppTraces(d.namespace, d.app!, params, d.cluster);
+        break;
+    }
```

Comment:
Maybe here it is better to add a simple if rather than switch since you are testing boolean statements (I thought it was the same case as the previous one). Additionally, avoid `any` if possible.

```suggestion
    let perfKey: string;
    let promise: Promise<ApiResponse<TracingResponse>>;
    if (d.workload) {
      perfKey = 'WorkloadTraces';
      promise = API.getWorkloadTraces(d.namespace, d.workload, params, d.cluster);
    } else if (d.service) {
      perfKey = 'ServiceTraces';
      promise = API.getServiceTraces(d.namespace, d.service, params, d.cluster);
    } else {
      perfKey = 'AppTraces';
      promise = API.getAppTraces(d.namespace, d.app!, params, d.cluster);
    }
```